### PR TITLE
e2e: ssl redirect

### PR DESCRIPTION
**Pull request by 3quanfeng**
1. re-organize test suites
2. add ssl-redirect test case

### CODE REVIEW

File: cmd/appgw-ingress/utils.go [hunk 1]
```diff
@@ -48,6 +48,12 @@ func validateNamespaces(namespaces []string, kubeClient *kubernetes.Clientset) e
 func getNamespacesToWatch(namespaceEnvVar string) []string {
 	// Returning an empty array effectively switches Ingress Controller
 	// in a mode of observing all accessible namespaces.
+	if len(namespaceEnvVar) != 0 {
+		glog.V(3).Infof("AGIC watchs namespace: %s", namespaceEnvVar)
```
Comment (akshaysngupta) [inline cmd/appgw-ingress/utils.go:52]: nit: `watches`

File: scripts/e2e/cmd/runner/runner_test.go [hunk 2]
```diff
@@ -54,7 +54,64 @@ var _ = Describe("Most frequenty run test suite", func() {
...
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			// clear all namespaces
+			cleanUp(clientset)
+		})
+	})
+
+	Context("one namespace many ingresses: thirty-four-ingresses-with-services", func() {
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/runner_test.go:90]: nit: fifty

File: scripts/e2e/cmd/runner/runner_test.go [hunk 3]
```diff
@@ -54,7 +56,64 @@ var _ = Describe("Most frequenty run test suite", func() {
...
+			Expect(err).To(BeNil())
+			Expect(publicIP).ToNot(Equal(""))
+
+			urlHttp = fmt.Sprintf("http://%s/status/200", publicIP)
+			urlHttps = fmt.Sprintf("https://%s/status/200", publicIP)
+		})
+
+		It("should get 200 ok for both http and https request", func() {
+			// http get to return 200 ok
+			err = makeGetRequest(urlHttp, "", 200, true)
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/runner_test.go:78]: if this is ssl-redirect, should we get a 3xx ?

File: cmd/appgw-ingress/utils.go [hunk 4]
```diff
@@ -48,6 +48,12 @@ func validateNamespaces(namespaces []string, kubeClient *kubernetes.Clientset) e
 func getNamespacesToWatch(namespaceEnvVar string) []string {
 	// Returning an empty array effectively switches Ingress Controller
 	// in a mode of observing all accessible namespaces.
+	if len(namespaceEnvVar) != 0 {
```
Comment (akshaysngupta) [inline cmd/appgw-ingress/utils.go:51]: We already have print statement for namespaces in the main.go file.
https://github.com/Azure/application-gateway-kubernetes-ingress/blob/ca2f88cec85dca87d072f68662fa556ddb9ced1a/cmd/appgw-ingress/main.go#L167

Comment (3quanfeng) [reply cmd/appgw-ingress/utils.go:51]: Thanks, removed

**Pull request merged**